src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -545,6 +545,9 @@ public function compile()
             if (!$definition->isPublic()) {
                 $this->privates[$id] = true;
             }
+            if ($definition->isSynthetic()) {
+                $this->synthetics[$id] = true;
```

Comment:
I know it was introduced this way for the `$privates` array, but to me, it doesn't make entirely sense.
Why not simply keep an array of synthetics services' ids instead of this map ?

Even if it won't happen, that's misleading, as we use `isset($this->privates/synthetics[$idToCheck])` later in the codebase, which returns `true` even if the value in the map is `false`.

A simple `in_array($idToCheck, $this->privates/synthetics)` would make more sense.